src/Instance/Move/useMoveItems.js

```diff
@@ -0,0 +1,57 @@
+import React, {
+  useState,
+  useContext,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import { CalloutContext } from '@folio/stripes-core';
+
+import { useMoveItemsMutation } from '../../common';
+import * as RemoteStorage from '../../RemoteStorageService';
+
+export const useMoveItems = () => {
+  const callout = useContext(CalloutContext);
+
+  const checkFromRemoteToNonRemote = RemoteStorage.Check.useByHoldings();
+
+  const [isMoving, setIsMoving] = useState(false);
+
+  const { mutate } = useMoveItemsMutation({
+    onMutate: () => setIsMoving(true),
+    onSettled:  () => setIsMoving(false),
+    onError: error => {
+      const { message } = error;
+      callout.sendCallout({ type: 'error', message });
+    },
+    onSuccess: (_, variables) => {
+      const message = (
+        <FormattedMessage
+          id="ui-inventory.moveItems.instance.items.success"
+          values={{ count: variables?.itemIds.length }}
```

Comment:
This just caught my eye because it looks incomplete. Either we know that `variables` and `itemIds` will always be there and we don't need the optional chaining at all or we need it completely with some nullish check.

```suggestion
          values={{ count: variables?.itemIds?.length ?? 0 }}
```